api/catalog/api/serializers/media_serializers.py

```diff
@@ -264,163 +167,263 @@ def validate(self, data):
+        help_text="The name of a detailed tag.",
     )
-    filetype = serializers.CharField(
+    accuracy = serializers.FloatField(
         required=False,
-        help_text="The type of the file, related to the file extension.",
+        help_text="The accuracy of a machine-generated tag. Human-generated "
+        "tags do not have an accuracy field.",
     )
 
-    # Fields corresponding to AbstractMedia
-    license = serializers.SerializerMethodField(
-        help_text="The name of license for the media."
+
+class MediaSearchSerializer(serializers.Serializer):
+    """
+    This serializer serializes the full media search response. The class should
+    be inherited by all individual media serializers.
+    """
+
+    result_count = serializers.IntegerField(
+        help_text="The total number of items returned by search result.",
     )
-    license_version = serializers.CharField(
-        required=False, help_text="The type of license for the media."
+    page_count = serializers.IntegerField(
+        help_text="The total number of pages returned by search result.",
     )
-    license_url = serializers.URLField(help_text="A direct link to the media license.")
-
-    provider = serializers.CharField(required=False, help_text="The content provider.")
-    source = serializers.CharField(
-        required=False,
-        help_text="The source of the data, meaning a particular dataset.",
+    page_size = serializers.IntegerField(
+        help_text="The number of items per page.",
     )
+    page = serializers.IntegerField(
+        help_text="The current page number returned in the response.",
+    )
+    # ``results`` field added by child serializers
+
+
+class MediaSerializer(BaseModelSerializer):
+    """
+    This serializer serializes a single media file. The class should be
+    inherited by all individual media serializers.
+    """
 
-    category = serializers.CharField(
-        required=True, allow_null=True, help_text="The category of the media."
+    class Meta:
+        model = AbstractMedia
+        fields = [
+            "id",
+            "title",
+            "foreign_landing_url",
+            "url",
+            "creator",
+            "creator_url",
+            "license",
+            "license_version",
+            "license_url",  # property
+            "provider",
+            "source",
+            "category",
+            "filesize",
+            "filetype",
+            "tags",
+            "attribution",  # property
+            "fields_matched",
+            "mature",
+        ]
+        """
+        Keep the fields names in sync with the actual fields below as this list is
+        used to generate Swagger documentation.
+        """
+
+    id = serializers.CharField(
+        help_text="Our unique identifier for an open-licensed work.",
+        source="identifier",
     )
 
     tags = TagSerializer(
-        allow_null=True,
+        allow_null=True,  # replaced with ``[]`` later
```

Comment:
What requirements are necessary for `later`? When can we do this replacement? 😄

Reply:
Later as in lower in the source code inside the `to_representation` function, not as in later in time.